### PR TITLE
SSCSSI-34: merge evidence share to tribunals api

### PR DESCRIPTION
**Pull request by ruban72**
### Jira link (if applicable)

[SSCSSI-34](https://tools.hmcts.net/jira/browse/SSCSSI-34)

### Change description ###
merge evidence share to tribunals api

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -96,6 +96,7 @@ tasks.withType(Test) {
 }
 
 task integration(type: Test) {
+    jvmArgs = ['-Xms1024m', '-Xmx2048m']
```
Comment (nitinprabhuhmcts) [inline build.gradle:99]: it works fine for me without jam args above not sure if it will work in the pipeline without those args ?

File: infrastructure/main.tf [hunk 2]
```diff
@@ -9,4 +9,31 @@ data "azurerm_key_vault" "sscs_key_vault" {
...
   azureVaultName = "sscs-${var.env}"
-}
\ No newline at end of file
+  sendgrid_subscription = {
+    prod    = "8999dec3-0104-4a27-94ee-6588559729d1"
+    nonprod = "1c4f0704-a29e-403d-b719-b90c34ef14c9"
+  }
+}
+
+data "azurerm_key_vault" "send_grid" {
```
Comment (nitinprabhuhmcts) [inline infrastructure/main.tf:18]: I think we should move the infra as a separate PR as we will already have the secrets in vault and build might fail. may be we will need to define secrets with new name but just a thought to do it as a separate PR ?

Comment (ruban72) [reply infrastructure/main.tf:18]: Sure. I will move these out to separate PR for infra

File: charts/sscs-tribunals-api/values.yaml [hunk 3]
```diff
@@ -77,10 +88,23 @@ java:
     ELINKS_V2_FEATURE_ENABLED: false
     DELETED_REDACTED_DOC_ENABLED: false
     UPLOAD_HEARING_FILTER_ENABLED: false
+    IDAM_OAUTH2_CLIENT_ID: "sscs"
+    CORE_CASE_DATA_JURISDICTION_ID: SSCS
+    CORE_CASE_DATA_CASE_TYPE_ID: Benefit
+    SEND_LETTER_SERVICE_BASEURL: "http://rpe-send-letter-service-{{ .Values.global.environment }}.service.core-compute-{{ .Values.global.environment }}.internal"
+    SEND_LETTER_SERVICE_ENABLED: "true"
+    TRUST_ALL_CERTS: "true"
```
Comment (nitinprabhuhmcts) [inline charts/sscs-tribunals-api/values.yaml:96]: just a general observation some env vars uses "" and some without "" may be we can make them consistent while you are here :)

**Review by nitinprabhuhmcts — APPROVED** [review]

**Review by nitinprabhuhmcts — APPROVED** [review]

**Review by Andrew-McMahon7 — DISMISSED** [review]

File: build.gradle [hunk 4]
```diff
@@ -373,6 +380,13 @@ dependencies {
     implementation group: 'org.apache.tomcat.embed', name: 'tomcat-embed-core', version: '9.0.86'
     implementation group: 'org.apache.tomcat.embed', name: 'tomcat-embed-websocket', version: '9.0.86'
 
+    //evidence-share depenedencies
+    implementation group: 'com.github.hmcts', name: 'cmc-pdf-service-client', version: '7.0.1'
+    implementation group: 'com.github.hmcts', name: 'send-letter-client', version: '3.0.23'
```
Comment (benouaer) [inline build.gradle:385]: might be a case for moving to EM document-generation with docmosis, rather than using the old pdf-service

File: charts/sscs-tribunals-api/values.preview.template.yaml [hunk 5]
```diff
@@ -14,4 +14,16 @@ java:
     CASE_DOCUMENT_AM_URL: http://ccd-case-document-am-api-aat.service.core-compute-aat.internal
     WORK_ALLOCATION_FEATURE: true
     CASE_ACCESS_MANAGEMENT_FEATURE: true
-    SCHEDULE_LISTING_FEATURE: true
\ No newline at end of file
+    SCHEDULE_LISTING_FEATURE: true
+    SEND_LETTER_SERVICE_ENABLED: "true"
```
Comment (benouaer) [inline charts/sscs-tribunals-api/values.preview.template.yaml:18]: are all these feature flags still needed?

File: src/main/java/uk/gov/hmcts/reform/sscs/evidenceshare/callback/handlers/ReissueFurtherEvidenceHandler.java [hunk 6]
```diff
@@ -0,0 +1,221 @@
...
+        ReissueArtifactUi reissueArtifactUi = sscsCaseData.getReissueArtifactUi();
+        reissueArtifactUi.setReissueFurtherEvidenceDocument(null);
+        reissueArtifactUi.setResendToAppellant(null);
+        reissueArtifactUi.setResendToRepresentative(null);
+        reissueArtifactUi.setResendToDwp(null);
+        reissueArtifactUi.setOtherPartyOptions(null);
+    }
+
+    @Override
+    public boolean canHandle(CallbackType callbackType, Callback<SscsCaseData> callback) {
```
Comment (benouaer) [inline src/main/java/uk/gov/hmcts/reform/sscs/evidenceshare/callback/handlers/ReissueFurtherEvidenceHandler.java:181]: this usually sits at the top, so it's clear what this handler is for

File: src/main/java/uk/gov/hmcts/reform/sscs/evidenceshare/service/CoverLetterService.java [hunk 7]
```diff
@@ -0,0 +1,162 @@
...
+        pdfsToBulkPrint.add(0, pdfCoverLetter);
+    }
+
+    /**
+     * Intended use of this method is only for local development or testing. This method produces a
+     * cover letter pdf file and stores it in the application root path.
+     *
+     * @param coverLetterContent this is the content in bytes of the cover letter from docmosis
+     */
+    private void printCoverLetterToPdfLocallyForDebuggingPurpose(byte[] coverLetterContent,
```
Comment (benouaer) [inline src/main/java/uk/gov/hmcts/reform/sscs/evidenceshare/service/CoverLetterService.java:62]: should this be continue to be shipped to production?

File: src/main/java/uk/gov/hmcts/reform/sscs/evidenceshare/service/EmailSenderProvider.java [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+    @Qualifier("sendGridMailSender")
+    private final JavaMailSender sendGridMailSender;
+
+    @Autowired
+    @Qualifier("mtaMailSender")
+    private final JavaMailSender mtaMailSender;
+
+
+    public JavaMailSender getMailSender() {
+        return featureToggleService.isSendGridEnabled() ? sendGridMailSender : mtaMailSender;
```
Comment (benouaer) [inline src/main/java/uk/gov/hmcts/reform/sscs/evidenceshare/service/EmailSenderProvider.java:25]: is the feature flag still needed?

File: src/main/java/uk/gov/hmcts/reform/sscs/evidenceshare/service/MockBulkPrintService.java [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+import uk.gov.hmcts.reform.sscs.ccd.domain.EventType;
+import uk.gov.hmcts.reform.sscs.ccd.domain.SscsCaseData;
+import uk.gov.hmcts.reform.sscs.docmosis.domain.Pdf;
+import uk.gov.hmcts.reform.sscs.evidenceshare.domain.FurtherEvidenceLetterType;
+import uk.gov.hmcts.reform.sscs.service.CcdNotificationsPdfService;
+
+
+@Service("bulkPrintService")
+@ConditionalOnProperty(prefix = "send-letter", name = "url", havingValue = "false")
+public class MockBulkPrintService implements PrintService {
```
Comment (benouaer) [inline src/main/java/uk/gov/hmcts/reform/sscs/evidenceshare/service/MockBulkPrintService.java:19]: should this be in the main code branch?

File: src/main/java/uk/gov/hmcts/reform/sscs/thirdparty/documentmanagement/DocumentStoreClient.java [hunk 10]
```diff
@@ -12,7 +12,7 @@
         name = "documentManagement",
         url = "${document_management.url}"
 )
-public interface DocumentManagementClient {
+public interface DocumentStoreClient {
     @DeleteMapping(value = "/documents/{documentId}")
```
Comment (benouaer) [inline src/main/java/uk/gov/hmcts/reform/sscs/thirdparty/documentmanagement/DocumentStoreClient.java:16]: should we still be talking to doc-store and not CDAM?

File: src/main/java/uk/gov/hmcts/reform/sscs/thirdparty/documentmanagement/DocumentStoreService.java [hunk 11]
```diff
@@ -44,6 +43,8 @@ public UploadResponse upload(List<MultipartFile> files) {
     }
 
     public void delete(String documentId) {
-        documentManagementClient.deleteDocument(OAUTH2_TOKEN, authTokenGenerator.generate(), USER_ID, documentId);
+        documentStoreClient.deleteDocument(OAUTH2_TOKEN, authTokenGenerator.generate(), USER_ID, documentId);
```
Comment (benouaer) [inline src/main/java/uk/gov/hmcts/reform/sscs/thirdparty/documentmanagement/DocumentStoreService.java:46]: should we still be talking to doc-store and not CDAM?

File: src/test/java/uk/gov/hmcts/reform/sscs/evidenceshare/service/MockBulkPrintServiceTest.java [hunk 12]
```diff
@@ -0,0 +1,49 @@
...
+import org.junit.runner.RunWith;
+import org.mockito.junit.MockitoJUnitRunner;
+import uk.gov.hmcts.reform.sscs.ccd.domain.EventType;
+import uk.gov.hmcts.reform.sscs.ccd.domain.SscsCaseData;
+import uk.gov.hmcts.reform.sscs.docmosis.domain.Pdf;
+import uk.gov.hmcts.reform.sscs.service.CcdNotificationsPdfService;
+
+
+@RunWith(MockitoJUnitRunner.class)
+public class MockBulkPrintServiceTest {
```
Comment (benouaer) [inline src/test/java/uk/gov/hmcts/reform/sscs/evidenceshare/service/MockBulkPrintServiceTest.java:20]: we are testing the mock service as well?

File: src/test/java/uk/gov/hmcts/reform/sscs/healthcheck/DocmosisHealthIndicatorTest.java [hunk 13]
```diff
@@ -0,0 +1,115 @@
...
+import org.springframework.boot.actuate.health.Health;
+import org.springframework.core.ParameterizedTypeReference;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.client.RestClientException;
+import org.springframework.web.client.RestTemplate;
+
+@RunWith(MockitoJUnitRunner.class)
+@SuppressWarnings("unchecked")
+public class DocmosisHealthIndicatorTest {
```
Comment (benouaer) [inline src/test/java/uk/gov/hmcts/reform/sscs/healthcheck/DocmosisHealthIndicatorTest.java:22]: are we testing docmosis health endpoint?

File: src/test/java/uk/gov/hmcts/reform/sscs/service/evidence/EvidenceUploadServiceTest.java [hunk 14]
```diff
@@ -137,7 +137,7 @@ public void setUp() {
 
     private void evidenceUploadService(AddedDocumentsUtil addedDocumentsUtil) {
         evidenceUploadService = new EvidenceUploadService(
-            mock(DocumentManagementService.class),
+            mock(DocumentStoreService.class),
```
Comment (benouaer) [inline src/test/java/uk/gov/hmcts/reform/sscs/service/evidence/EvidenceUploadServiceTest.java:140]: should we still be talking to doc-store and not CDAM?

**Review by benouaer — DISMISSED** [review]

**Review by nitinprabhuhmcts — APPROVED** [review]

**Review by Andrew-McMahon7 — APPROVED** [review]
